src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1102,6 +1111,21 @@ protected async Task<TResult> InvokeCoreAsync<TResult>(long? id, string targetNa
                 request.Arguments = arguments ?? EmptyObjectArray;
             }
 
+            return await this.InvokeCoreAsync<TResult>(request, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected async Task<TResult> InvokeCoreAsync<TResult>(JsonRpcRequest request, CancellationToken cancellationToken)
+        {
+            long id = default(long);
```

Comment:
In the event that `request.Id is null`, this will leave `id=0`, which is *not* the same thing as `null`. Being `null` implies a notification while `0` is a request that expects a response.
Also, in the general relaying configuration, you can't assume that the `request.Id` is even a number. It's allowed to be `string`, and some JSON-RPC servers do that. So if we're relaying messages from non-StreamJsonRpc sources, we'd need to properly handle all legal values for id.